nlp/gender_debiasing_cda/gender_debiasing_cda.ipynb

```diff
@@ -0,0 +1,1032 @@
+      },
+      "source": [
+        "### Imports"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 1,
+      "metadata": {
+        "id": "iLiKoNuq1MLn"
+      },
+      "outputs": [],
+      "source": [
+        "import numpy as np\n",
+        "from scipy import spatial\n",
+        "import multiprocessing\n",
+        "from datetime import datetime\n",
+        "import pandas as pd\n",
+        "\n",
+        "import plotly.graph_objects as go\n",
+        "\n",
+        "import augly.text as textaugs\n",
+        "from gender_bender import gender_bend\n",
+        "from sklearn.decomposition import PCA\n",
+        "\n",
+        "from datasets import concatenate_datasets, Dataset, load_from_disk"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ipA94F-h01Px"
+      },
+      "source": [
+        "### Download dataset\n",
+        "\n",
+        "We'll train our word vectors on the [Wiki-Bio](https://rlebret.github.io/wikipedia-biography-dataset/) dataset, which is a collection various biography pages from Wikipedia. This seems like an ideal candidate for our gender-bias experiment.\n",
+        "\n",
+        "We pre-downloaded the dataset from the Github page. If you are unable to do so, feel free to just load in our preprocessed dataset in the section **Load in the data**."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "CdIlM9fPT4SS"
+      },
+      "outputs": [],
+      "source": [
+        "wiki_bio_sent = []\n",
+        "wiki_bio_nb = []\n",
+        "wiki_bio_title = []\n",
+        "wiki_bio_id = []\n",
+        "\n",
+        "with open('wikipedia-biography-dataset/test/test.sent', 'r') as f:\n",
+        "    wiki_bio_sent = f.read().splitlines() \n",
+        "\n",
+        "with open('wikipedia-biography-dataset/test/test.nb', 'r') as f:\n",
+        "    wiki_bio_nb = f.read().splitlines()\n",
+        "\n",
+        "with open('wikipedia-biography-dataset/test/test.title', 'r') as f:\n",
+        "    wiki_bio_title = f.read().splitlines() \n",
+        "\n",
+        "with open('wikipedia-biography-dataset/test/test.id', 'r') as f:\n",
+        "    wiki_bio_id = f.read().splitlines() "
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "zvu9vWdiT4SZ"
+      },
+      "source": [
+        "It's in a bit of a weird format, so we'll have to merge it together"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "KldSaS8bT4Sc"
+      },
+      "outputs": [],
+      "source": [
+        "i = 0\n",
+        "wiki_bio_sent_grouped = []\n",
+        "for nr in wiki_bio_nb:\n",
+        "    nr_int = int(nr)\n",
+        "    wiki_sent_slice = wiki_bio_sent[i:i+nr_int]\n",
+        "    wiki_bio_sent_grouped.append(' '.join(wiki_sent_slice))\n",
+        "    i+=nr_int"
```

Comment:
inconsistent whitespace handling between variables and operators, e.g. `a=b` versus `a = b`. This inconsistency spans over multiple notebook cells

Reply:
Good catch, unified the spacing